versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -194,7 +196,8 @@ protected CommitLogEntry commitAttempt(
             commitAttempt.getPuts(),
             commitAttempt.getDeletes(),
             currentBranchEntry != null ? currentBranchEntry.getKeyListDistance() : 0,
-            newKeyLists);
+            newKeyLists,
+            new LinkedList<>());
```

Comment:
Why LinkedList? It's a plain list - ArrayList is more than sufficient.

Reply:
wanted to maintain the order, I can keep a list and insert at head aswell.

Reply:
The plain `List` interface definition implies ordered entries too.